lib/Basic/Statistic.cpp

```diff
@@ -458,7 +462,7 @@ UnifiedStatsReporter::printAlwaysOnStatsAndTimers(raw_ostream &OS) {
     auto &C = getFrontendCounters();
 #define FRONTEND_STATISTIC(TY, NAME)                          \
     do {                                                      \
-      if (C.NAME) {                                           \
+      if (C.NAME || IsPrintingZeroStats) {                    \
```

Comment:
You'll also want to check for this in `publishAlwaysOnStatsToLLVM` for the `Stat = 0` assignment to ensure consistent behaviour across debug and release builds